### ISSUE TITLE
Support more patterns

### ISSUE THREAD

**Issue by wildskyf** [wildskyf/personal-blocklist#36]
Current rules is 'if url matches any pattern in list, block it'.

I want it support Regex (or wildcard), title match, ... etc

**Comment by taxen** [comment]
I push this feature as I see lots of junk domains including *.tk, *.cf, *.ml, *.gq, *ga etc.

**Comment by wildskyf** [comment]
Currently, you could add `.tk`, `.cf`, `.ml`, `.gq`, `.ga` in order to  block those websites from search results.

But it is still better to have this feature. :)

**Comment by taxen** [comment]
Yes, I figured it out. Instead of *.tk, it is better to use '.tk/' with spash so that the pattern only matches when 'tk' comes just before slash. Otherwise, '.tk' also matches '.tking' or anything with 'tk' at the beginning. Still, '.tk/' does not assure this part belongs to top domain, but could belong any part of a URL. I hope https?://*.tk/ type regex will be implemented some day.

**Comment by taxen** [comment]
Yes, I figured it out. Instead of .tk, it is better to use '.tk/' with slash so that the pattern only matches when 'tk' comes just before slash. Otherwise, '.tk' also matches '.tking' or anything with 'tk' at the beginning. Still, '.tk/' does not assure this part belongs to top domain, but could belong any part of a URL. I hope https?://.tk/ type regex will be implemented some day.
